scripts/loc.sh

```
@@ -0,0 +1,183 @@
+total_all_lines=0
+
+# Bash 3.x compatible arrays (using parallel arrays instead of associative)
+test_lines_by_type=()
+nontest_lines_by_type=()
+total_lines_by_type=()
+file_type_names=()
+
+# Calculate lines for each file type
+for i in "${!FILE_TYPES[@]}"; do
+    ext="${FILE_TYPES[$i]}"
+    read test_count nontest_count <<< $(count_test_vs_nontest "$ext")
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)